src/main/java/com/esri/samples/na/offline_routing/OfflineRoutingSample.java

```diff
@@ -97,6 +99,15 @@ public void start(Stage stage) throws Exception {
       routeTask = new RouteTask("./samples-data/san_diego/sandiego.geodatabase", "Streets_ND");
       routeTask.loadAsync();
 
+      // add a graphics overlay to show the boundary
+      Envelope envelope = new Envelope(new Point(-13045352.223196, 3864910.900750, 0, SpatialReferences
+      .getWebMercator()), new Point(-13024588.857198, 3838880.505604, 0, SpatialReferences.getWebMercator()));
+      SimpleLineSymbol boundarySymbol = new SimpleLineSymbol(SimpleLineSymbol.Style.DASH, 0xFF00FF00, 5);
+      Graphic boundary = new Graphic(envelope, boundarySymbol);
+      GraphicsOverlay boundaryOverlay = new GraphicsOverlay();
+      boundaryOverlay.getGraphics().add(boundary);
+      mapView.getGraphicsOverlays().add(boundaryOverlay);
+
       // create route parameters
       routeParameters = routeTask.createDefaultParametersAsync().get();
```

Comment:
On line 197 i would check that the route task is loaded before doing, `routeTask.getRouteTaskInfo().getTravelModes()`. Application will crash if task not loaded before accessing RouteTaskInfo.